tests/mr/device/adaptor_tests.cpp

```diff
@@ -64,6 +66,30 @@ using adaptors = ::testing::Types<aligned_resource_adaptor<cuda_mr>,
                                   thread_safe_resource_adaptor<cuda_mr>,
                                   tracking_resource_adaptor<cuda_mr>>;
 
+static_assert(
+  cuda::mr::resource_with<rmm::mr::aligned_resource_adaptor<cuda_mr>, cuda::mr::device_accessible>,
```

Comment:
What does `cuda::mr::resource_with<>` do? Does it return bool? (The name is non-obvious.)

Reply:
This is a C++20 concept. So what it does it returns true if the type passed in satisfies the `resource` concept and any of the additional constraints passed in (in this case `cuda::mr::device_accessible`).

Otherwise it is false. As @jrhemstad pointed out this is the conventional way of checking whether a type supports a concepts.

In templates you will then find things like
```cpp
template <cuda::mr::resource_with<cuda::mr::device_accessible> Resource>
void* some_function(...) {....}
```

This is equivalent to
```cpp
template <class Resource>
    requires cuda::mr::resource_with<Resource, cuda::mr::device_accessible>
void* some_function(...) {....}
```

Reply:
So clearly this works without concepts since we are compiling for C++17?

Reply:
Yeah, we implemented a fallback mechanism that emulates concept behavior in C++14 / c++17

Here this act like a `constexpr bool` variable